docs/docs/queries/structure.md

```diff
@@ -1,44 +1,207 @@
+Let's have a look at how  we can put DQL to use.
 
-1. **TABLE**: The traditional view type; one row per data point, with several columns of field data.
-2. **LIST**: A list of pages which match the query. You can output a single associated value for each page.
-3. **TASK**: A list of tasks whose pages match the given query.
-4. **CALENDAR**: A calendar view displaying each hit via a dot on its referred date
+## General Format of a DQL Query
 
-Read more about the available types [here](./query-types.md)
+Every query follows the same structure and consists of
 
-## General Format
+- exactly one **Query Type** with zero, one or many [fields](../annotation/add-metadata.md), depending on query type.
+- zero or one **FROM** data commands with one to many [sources](../reference/sources.md)
+- zero to many other **data commands** with one to many [expressions](../reference/expressions.md) and/or other infos depending on the data command 
 
-The general format for queries is:
+At a high level, a query conforms to the following pattern:
 
 ~~~
 ```dataview
-TABLE|LIST|TASK <field> [AS "Column Name"], <field>, ..., <field> 
+<QUERY-TYPE> <fields>
 FROM <source>
-WHERE <expression>
-SORT <expression> [ASC/DESC]
-... other data commands
+<DATA-COMMAND> <expression>
+<DATA-COMMAND> <expression>
+          ...
+```
+~~~
+
+!!! hint "Only the Query Type is mandatory."
+
+The following sections will explain the theory in further detail.
+
+## Choose a Output Format
+
+The output format of a query is determined by its **Query Type**. There are four available:
+
+1. **TABLE**: A table of results with one row per result and one to many columns of field data.
+2. **LIST**: A bullet point list of pages which match the query. You can output one field for each page alongside their file links.
+3. **TASK**: An interactive task list of tasks that match the given query.
+4. **CALENDAR**: A calendar view displaying each hit via a dot on its referred date.
+
+The Query Type is the **only mandatory command in a query**. Everything else is optional.
+
+!!! attention "Possibly memory expensive examples"
+    Depending on the size of your vault, executing the following examples can take long and even freeze Obsidian in extreme cases. It's recommended that you specify a `FROM` to restrict the query execution to a specific subset of your vaults' files. See next section.  
+
+~~~
+Lists all pages in your vault as a bullet point list
+```dataview
+LIST
+```
+
+Lists all tasks (completed or not) in your vault
+```dataview
+TASK
+```
+
+Renders a Calendar view where each page is represented as a dot on its creation date.
+```dataview
+CALENDAR file.cday
+```
+
+Shows a table with all pages of your vault, their field value of due, the files' tags and an average of the values of multi-value field working-hours
+```dataview
+TABLE due, file.tags AS "tags", average(working-hours)
+```
+~~~
+
+!!! info "Read more about the available types and how to use them [here](./query-types.md)."
```

Comment:
What types? Query types may be readily apparent given where we are in the docs but again, there's potential to confuse this with data types. More explicit wording would be helpful.